jscomp/others/js_dict.ml

```diff
@@ -56,3 +56,55 @@ let unsafeDeleteKey : string t -> string -> unit [@bs] = [%raw{|
      return 0
    }
 |}]
+
+(* external entries : 'a t -> (key * 'a) array = "Object.entries" [@@bs.val] (* ES2017 *) *)
+let entries dict =
+  let keys = keys dict in
+  let l = Js.Array.length keys in
+  (* TODO: Get rid of Obj.magic *)
+  let values = Obj.magic (Array.make l 0) in
+  for i = 0 to l - 1 do
+    let key = Array.unsafe_get keys i in
+    Array.set values i (key, unsafeGet dict key)
+  done;
+  values
+
+(* external values : 'a t -> 'a array = "Object.values" [@@bs.val] (* ES2017 *) *)
+let values dict =
```

Comment:
They are also not widely supported in the browser (at least if one does have to support older browsers).